libsql-server/src/replication/snapshot.rs

```diff
@@ -178,7 +182,12 @@ fn pending_snapshots_list(compact_queue_dir: &Path) -> anyhow::Result<Vec<(LogFi
 }
 
 impl LogCompactor {
-    pub fn new(db_path: &Path, log_id: Uuid, callback: SnapshotCallback) -> anyhow::Result<Self> {
+    pub fn new(
+        db_path: &Path,
+        log_id: Uuid,
+        callback: SnapshotCallback,
+        encryption: Option<FrameEncryptor>,
```

Comment:
yeah, it only encrypts/decrypts because it reuses the same code (`into_stream_mut`) that's also used to send data over the wire, and that gets sent unencrypted. But that can be solved by using a SnapshotFile instance without the encryption key set I guess... I'll play with it to make it work without excessive encryption/decryption passes

Reply:
yeah it looks like the snapshots already not encrypt headers, and it's just a plain bug that I try to decrypt both the page and header. I need another pass on this snapshotting code